cardano-crypto-class/src/Cardano/Crypto/DSIGN/Ed25519.hs

```diff
@@ -6,6 +6,17 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE TypeApplications #-}
+-- According to the documentation for unsafePerformIO:
+-- 
+-- > Make sure that the either you switch off let-floating 
+-- > (-fno-full-laziness), or that the call to unsafePerformIO cannot float 
+-- > outside a lambda.
+--
+-- If we do not switch off let-floating, our calls to unsafeDupablePerformIO for
+-- FFI functions become nondeterministic in their behaviour when run with
+-- parallelism enabled (such as -with-rtsopts=-N), possibly yielding wrong
+-- answers on a range of tasks, including serialization.
```

Comment:
There are a number of other places in this package where we use unsafe IO functions, we should check those also! :scream: 

cc @tdammers

Reply:
The tests don't trigger any non-determinism when run in parallel now, even with increased test counts. A closer look would be good, but I suspect with all the runs I've done, we'd have caught something by now.

Reply:
Yeah, but IMO the use sites could do with auditing and a justification for why it's okay.